log4j-core/src/main/java/org/apache/logging/log4j/core/LoggerContext.java

```diff
@@ -91,6 +93,7 @@ public class LoggerContext extends AbstractLifeCycle
     private String contextName;
     private volatile URI configLocation;
     private Cancellable shutdownCallback;
+    private MessageFactory defaultMessageFactory = ParameterizedMessageFactory.INSTANCE;
```

Comment:
The default message factory here should depend on `log4j2.messageFactory` and `log4j2.enableThreadlocals`.

Reply:
I think I would prefer to call `Logger.getEffectiveMessageFactory(null)` (which is in the same package).

Instantiating a logger just to see what factory will it use, seems like a big workaround.

Reply:
Unless you strongly object to it, I want to stick with the _big workaround_ approach.